### PR TITLE
Feature/unit tests/infra 30038/cim tests

### PR DESCRIPTION
**Pull request by kkania-splunk**
unit tests for standard_lib/cim_tests

### CODE REVIEW

**Comment by kkania-splunk** [comment]
tests for `test_templates` are omitted intentionally, it has many interactions with internal `pytest` structure and has good coverage by 'functional' tests. If needed, can be covered after other modules done.

File: tests/unit/tests_standard_lib/test_cim_tests/test_field_test_helper.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+                },
+            ],
+        ),
+    ],
+)
+def test_parse_result(
+    field_test_adapter_mock, mocked_field_test_helper, results, fields, expected_output
+):
+    mocked_field_test_helper.fields = fields or []
+    if fields:
```
Comment (uoboda-splunk) [inline tests/unit/tests_standard_lib/test_cim_tests/test_field_test_helper.py:185]: Probably that can be avoided by defining global vars:
```
field_1 = field(name="field_1", gen_validity_query=lambda: True)
field_2 = field(name="field_2", gen_validity_query=lambda: True)
```
And using it in parametrisation directly

File: tests/unit/tests_standard_lib/test_cim_tests/test_field_test_helper.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            "valid_field_count": 1,
+            "invalid_values": ["null", "none"],
+        },
+    ]
+    mocked_field_test_helper.fields = fields or []
+    mocked_field_test_helper.search = "base_search"
+    new_line = "\n"
+    with patch.object(
+        FieldTestHelper,
+        "get_table_output",
```
Comment (uoboda-splunk) [inline tests/unit/tests_standard_lib/test_cim_tests/test_field_test_helper.py:288]: that lambda is a bit complex and hard to understand. Maybe define normal function to make it more clear

**Review by uoboda-splunk — APPROVED** [review]

**Pull request merged**